Remotion/Core/Core/Configuration/TypeDiscovery/TypeDiscoveryConfiguration.cs

```diff
@@ -55,7 +55,7 @@ public static void SetCurrent (TypeDiscoveryConfiguration configuration)
 
     private static TypeDiscoveryConfiguration GetTypeDiscoveryConfiguration ()
     {
-      return (TypeDiscoveryConfiguration) 
+      return (TypeDiscoveryConfiguration?)

```

Comment:
the cast happens here before the ??-operator

Reply:
That, I understand. But the method returns `TypeDiscoveryConfiguration` not `TypeDiscoveryConfiguration?`.

Reply:
the method cant return null 
if the 1st part of the ?? is null it returns `new TypeDiscoveryConfiguration()`

Reply:
I know. But why cast to `TypeDiscoveryConfiguration?` (nullable) then? Seems like roslyn knows that it will never be `null` though, as it wouldn't compile otherwise.

Reply:
the cast is only applied to `ConfigurationWrapper.Current.GetSection ("remotion.typeDiscovery", false)`, which returns something nullable

Reply:
```suggestion
      return ((TypeDiscoveryConfiguration)
          ConfigurationWrapper.Current.GetSection ("remotion.typeDiscovery", false) ?? new TypeDiscoveryConfiguration());
```
would be also correct

Reply:
...my bad. confused this with the change in `Remotion/Core/Core/Configuration/ServiceLocation/ServiceLocationConfiguration.cs` looking the same but having parethesis. LOL